### PR TITLE
chore: 🧑‍💻 add `devtest` environment for quick development tests

### PR DESCRIPTION
**Pull request by zoido**
Current `test` environment has matrix for `httpx` and `python` versions and tests against packaged version of the project to verify the build result.
This is overkill for local cevelopment sanity checks. Hatch does not update the project when there are changes to the sources so `dev-mode = false` `hatch env remove` is neeede with `test` ro verify changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,3 +64,33 @@ async def serve(q: Q):
...
 
 ```
+
+## Development
+
+Project is managed using [Hatch](https://hatch.pypa.io/latest/).
+
+### Testing
+
+For quick development tests use :
```
Comment (tomasz-h2o) [inline README.md:74]: nitty nit: whitespace before `:`

**Review by tomasz-h2o — DISMISSED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (zoido) [reply README.md:74]:
```suggestion
For quick development tests use:
```

Comment (zoido) [reply README.md]: Thanks.

**Comment by zoido** [comment]
@tomasz-h2o PTAL. Dismissing reviews seemed to be turned on on the branch protection.

**Review by tomasz-h2o — APPROVED** [review]

**Pull request merged**